File: src/java/org/codehaus/testdox/intellij/TestDoxNonProjectClass.java
package org.codehaus.testdox.intellij;

import javax.swing.Icon;

import com.intellij.openapi.vfs.VirtualFile;

import org.codehaus.testdox.intellij.panel.TestDoxModel;

public class TestDoxNonProjectClass extends TestDoxFile {

    public static final TestElement TEST_ELEMENT = new AbstractTestElement() {
        public String displayString() {
            return "<font color=\"gray\">Not in current project</font>";
        }

        public Icon icon() {
            return IconHelper.getIcon(IconHelper.WARNING_ICON);
        }
    };

    public TestDoxNonProjectClass(VirtualFile file, String className, TestClass testClass, TestClass testedClass) {
        super(file, className, testClass, testedClass, TestMethod.EMPTY_ARRAY);
    }

    public void updateModel(TestDoxModel model) {
        model.setTestDoxForNonProjectClass(this);
    }
}

File: src/scala/org/codehaus/testdox/intellij/TestDoxNonProjectClass.scala
package org.codehaus.testdox.intellij

import com.intellij.openapi.vfs.VirtualFile

import org.codehaus.testdox.intellij.ui.TestDoxTableModel

class TestDoxNonProjectClass(file: VirtualFile, className: String, testClass: TestClass, testedClass: TestClass)
    extends TestDoxFile(file, className, testClass, testedClass, TestMethod.EMPTY_ARRAY) {

  def updateModel(model: TestDoxTableModel) {
    model.setTestDoxForNonProjectClass(this)
  }
}

object TestDoxNonProjectClass {

  val TEST_ELEMENT = new AbstractTestElement() {
    val displayString = "<font color=\"gray\">Not in current project</font>"
    val icon = IconHelper.getIcon(IconHelper.WARNING_ICON)
  }
}
